hazelcast/src/main/java/com/hazelcast/mapreduce/impl/task/DefaultContext.java

```diff
@@ -102,11 +102,12 @@ public void finalizeCombiners() {
         Combiner<ValueIn, ?> combiner = combiners.get(key);
         if (combiner == null) {
             combiner = combinerFactory.newCombiner(key);
```

Comment:
That'll work too, provided the implementation is actually able to do this inside the lock.